compile error: sys/timeb.h not found

It seems that in the android-ndk-r10, the file timeb.h is removed, which leads to a compiling failure.
Has no idea of how to fix it besides downloading a r9 version of ndk...

P.S.: And in r10, the API dir has been changed into Android-L from Android-14 or so, which I made a link to solve the problem.

Thanks for any suggestion.



<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/3936418-compile-error-sys-timeb-h-not-found?utm_campaign=plugin&utm_content=tracker%2F11371&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F11371&utm_medium=issues&utm_source=github).
</bountysource-plugin>

The error message that appears is:

```make
ccache arm-linux-androideabi-gcc -DANDROID -mandroid  -fomit-frame-pointer --sysroot /home/samo/Dev/android-ndk-r10d/platforms/android-21/arch-arm -fno-strict-aliasing -DANDROID -mandroid  -fomit-frame-pointer --sysroot /home/samo/Dev/android-ndk-r10d/platforms/android-21/arch-arm -DNDEBUG   -I. -IInclude -I./Include  -fPIC -DPy_BUILD_CORE  -I/home/samo/Dev/python-for-android/build/python-install/include -c ./Modules/zlibmodule.c -o Modules/zlibmodule.o
./Modules/timemodule.c:31:23: fatal error: sys/timeb.h: No such file or directory
 #include <sys/timeb.h>
                       ^
compilation terminated.
make: *** [Modules/timemodule.o] Error 1
```
Which is required here:

https://github.com/python/cpython/blob/2.2/Modules/timemodule.c#L28

but not in the latest version, there seems to be a workaround here:

https://github.com/python/cpython/blob/2.7/Modules/timemodule.c#L919

After checking /android-ndk-r10d/platforms/android-21/arch-arm/usr/include/sys/, timeb.h is indeed missing in API 21, but it's available in API 19 and lower. This looks like a bad combination of the latest API and Python 2.7.2.

--- a/pythonforandroid/recipes/python2/__init__.py
+++ b/pythonforandroid/recipes/python2/__init__.py
@@ -11,7 +11,7 @@ class Python2Recipe(Recipe):
     url = 'http://python.org/ftp/python/{version}/Python-{version}.tar.bz2'
     name = 'python2'
 
-    depends = ['hostpython2']  
+    depends = ['hostpython2']
     conflicts = ['python3']
 
     def prebuild_armeabi(self):
@@ -38,6 +38,9 @@ class Python2Recipe(Recipe):
         if uname()[0] == 'Linux':
             self.apply_patch(join('patches', 'fix-configure-darwin.patch'))
             self.apply_patch(join('patches', 'fix-distutils-darwin.patch'))
+
+        if self.ctx.android_api > 19:
+            self.apply_patch(join('patches', 'fix-ftime-removal.patch'))
 
         shprint(sh.touch, join(build_dir, '.patched'))
 
@@ -125,7 +128,7 @@ class Python2Recipe(Recipe):
                         _env=env)
             except sh.ErrorReturnCode_2:
                 print('First python2 make failed. This is expected, trying again.')
-                
+
 
             print('Second install (expected to work)')
             shprint(sh.touch, 'python.exe', 'python')

--- a/pythonforandroid/recipes/python2/patches/fix-ftime-removal.patch
+++ b/pythonforandroid/recipes/python2/patches/fix-ftime-removal.patch
@@ -1,0 +1,11 @@
+diff -u -r Python-2.7.2.orig/Modules/timemodule.c Python-2.7.2/Modules/timemodule.c
+--- Python-2.7.2.orig/Modules/timemodule.c	2011-06-11 15:46:27.000000000 +0000
++++ Python-2.7.2/Modules/timemodule.c	2015-09-11 10:37:36.708661691 +0000
+@@ -27,6 +27,7 @@
+ #include <io.h>
+ #endif
+ 
++#undef HAVE_FTIME
+ #ifdef HAVE_FTIME
+ #include <sys/timeb.h>
+ #if !defined(MS_WINDOWS) && !defined(PYOS_OS2)

